src/imgui/ImGuiReverseBar.cc

```diff
@@ -62,23 +62,26 @@ void ImGuiReverseBar::showMenu(MSXMotherBoard* motherBoard)
 		}
 		ImGui::Separator();
 
-		im::Menu("Load state ...", [&]{
+		loadStateOpen = im::Menu("Load state ...", [&]{
+			if (!loadStateOpen) {
```

Comment:
Is this correct?  (I haven't tested myself.)

I think `im::Menu()` returns `true` when the menu is _currently_ open. You want to know when it changed from closed to open. So I think your current code will never execute.

Maybe this could be a solution:
```
bool isOpen = im::Menu(..., [&]{
    if (!wasOpen) {
        wasOpen = isOpen;
        ...
    });
```
Another way to see that there's something fishy going on is:
* In this function you unconditionally overwrite `loadStateOpen` before using it.
* And this function is the only user of `loadStateOpen`.
* So in the current structure it wasn't needed to "remember" it (= make it a member variable).
* But we do need to compare state from the previous frame with this frame, so we do need some "memory".

Reply:
Um, well, it did seem to work and in fact, I thought I was using the same construction as already present in the code, see https://github.com/openMSX/openMSX/blob/bb18df8c864469563db24f95ea55a0baeee0db8f/src/imgui/ImGuiReverseBar.cc#L192C9-L192C23 in combination with https://github.com/openMSX/openMSX/blob/bb18df8c864469563db24f95ea55a0baeee0db8f/src/imgui/ImGuiReverseBar.cc#L186

Reply:
Aha. I understand now. I was wrong.

The assignment to 'loadStateOpen' only happens _after_ the lambda has been executed. So inside that lambda you still see the "old" value of this variable.

Reply:
So, please resolve the conversation if you're happy how things are (after all).